cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js

```diff
@@ -303,45 +302,52 @@ function cancelOrRefund() {
  * 
  * @returns redering template or error
  */
-function authorizeWithForm()
-{
-	var	adyen3DVerification = require('int_adyen_overlay/cartridge/scripts/adyen3DVerification'), result,
-	order = session.custom.order,
-	paymentInstrument = session.custom.paymentInstrument,
-	adyenResponse  = session.custom.adyenResponse;
+function authorizeWithForm() {
+	var order = session.custom.order,
+		paymentInstrument = session.custom.paymentInstrument,
+		adyenResponse = session.custom.adyenResponse;
+
 	clearCustomSessionFields();
-	
 	Transaction.begin();
-	result = adyen3DVerification.verify({
-		Order: order,
-		Amount: paymentInstrument.paymentTransaction.amount,
-		PaymentInstrument: paymentInstrument,
-		CurrentSession: session,
-		CurrentRequest: request,
-		MD: adyenResponse.MD,
-		PaResponse: adyenResponse.PaRes,
-		PaymentData : paymentInstrument.custom.adyenPaymentData
-	});
-	
-    if (result.error || result.Decision != 'ACCEPT') {
-    	Transaction.rollback();
-    	Transaction.wrap(function () {
-            paymentInstrument.custom.adyenPaymentData = null; 
+	var adyenCheckout = require('int_adyen_overlay/cartridge/scripts/adyenCheckout');
+	var jsonRequest = {
+		"paymentData": paymentInstrument.custom.adyenPaymentData,
+		"details": {
+			"MD": adyenResponse.MD,
+			"PaRes": adyenResponse.PaRes
+		}
+	};
+
+	var result = adyenCheckout.doPaymentDetailsCall(jsonRequest);
+
+	if (result.error || result.resultCode != 'Authorised') {
+		Transaction.rollback();
+		Transaction.wrap(function () {
+			paymentInstrument.custom.adyenPaymentData = null;
 			OrderMgr.failOrder(order);
 		});
 		app.getController('COSummary').Start({
-            PlaceOrderError: new Status(Status.ERROR, 'confirm.error.declined', '')
-        });
+			PlaceOrderError: new Status(Status.ERROR, 'confirm.error.declined', '')
+		});
 		return;
-    }
-    
+	}
+	if ('pspReference' in result && !empty(result.pspReference)) {
+		paymentInstrument.paymentTransaction.transactionID = result.pspReference; 
+		order.custom.Adyen_pspReference = result.pspReference;
+	}
+	if ('resultCode' in result && !empty(result.resultCode)) {
+		paymentInstrument.paymentTransaction.custom.authCode = result.resultCode;
+	}
+
+	// Save full response to transaction custom attribute
+	paymentInstrument.paymentTransaction.custom.Adyen_log = JSON.stringify(result);
```

Comment:
I know we do it on more places but what is the purpose of adding the full response into the DB rather then log it.